TST: tolerance violation in `test_x0_working[tfqmr]` on windows

While testing main for https://github.com/scipy/scipy/pull/21982 in conda-forge, I keep running into 

```
=========================== short test summary info ===========================
FAILED sparse/linalg/_isolve/tests/test_iterative.py::test_x0_working[tfqmr] - assert np.float64(4.970321428218439e-06) <= (3e-06 * np.float64(1.240298651028214))
 +  where np.float64(4.970321428218439e-06) = norm(((array([[3.68514493, 2.22433683, 2.80424149, 2.06524296, 2.82772901,\n        1.70134349, 2.98806253, 2.88379072, 1.9578..., 2.29515724, 2.77591422, 1.55001596, 1.93969785,\n        1.88091968, 2.24894423, 2.49791763, 1.66944017, 2.12310297]]) @ array([ -1.22141228, -23.52876153,  -4.72398842, -21.36849916,\n       -12.68970175,  -0.96738895,  32.85873013, -11.40287582,\n        10.99363035,  30.93079899])) - array([0.55127821, 0.22828434, 0.09135873, 0.07249963, 0.11693987,\n       0.74246847, 0.72216603, 0.1607452 , 0.15471589, 0.18022921])))
 +  and   np.float64(1.240298651028214) = norm(array([0.55127821, 0.22828434, 0.09135873, 0.07249963, 0.11693987,\n       0.74246847, 0.72216603, 0.1607452 , 0.15471589, 0.18022921]))
= 1 failed, 73700 passed, 4669 skipped, 325 xfailed, 17 xpassed, 2593 warnings in 1360.58s (0:22:40) =
```
for about 50% of the runs on windows.

We've already loosened the tolerance for this tests in the 1.15 cycle (#20952), but it seems that something recently caused another degradation (this doesn't necessarily mean it happened in SciPy, could also be in the BLAS stack in conda-forge, which has its own set of issues, e.g. MKL 2025.0 has a bunch of regressions).

<details>

```
================================== FAILURES ===================================
___________________________ test_x0_working[tfqmr] ____________________________
[gw0] win32 -- Python 3.13.1 %PREFIX%\python.exe
..\_test_env\Lib\site-packages\scipy\sparse\linalg\_isolve\tests\test_iterative.py:508: in test_x0_working
    assert norm(A @ x - b) <= 3e-6*norm(b)
E   assert np.float64(4.970321428218439e-06) <= (3e-06 * np.float64(1.240298651028214))
E    +  where np.float64(4.970321428218439e-06) = norm(((array([[3.68514493, 2.22433683, 2.80424149, 2.06524296, 2.82772901,\n        1.70134349, 2.98806253, 2.88379072, 1.9578..., 2.29515724, 2.77591422, 1.55001596, 1.93969785,\n        1.88091968, 2.24894423, 2.49791763, 1.66944017, 2.12310297]]) @ array([ -1.22141228, -23.52876153,  -4.72398842, -21.36849916,\n       -12.68970175,  -0.96738895,  32.85873013, -11.40287582,\n        10.99363035,  30.93079899])) - array([0.55127821, 0.22828434, 0.09135873, 0.07249963, 0.11693987,\n       0.74246847, 0.72216603, 0.1607452 , 0.15471589, 0.18022921])))
E    +  and   np.float64(1.240298651028214) = norm(array([0.55127821, 0.22828434, 0.09135873, 0.07249963, 0.11693987,\n       0.74246847, 0.72216603, 0.1607452 , 0.15471589, 0.18022921]))
        A          = array([[3.68514493, 2.22433683, 2.80424149, 2.06524296, 2.82772901,
        1.70134349, 2.98806253, 2.88379072, 1.9578..., 2.29515724, 2.77591422, 1.55001596, 1.93969785,
        1.88091968, 2.24894423, 2.49791763, 1.66944017, 2.12310297]])
        b          = array([0.55127821, 0.22828434, 0.09135873, 0.07249963, 0.11693987,
       0.74246847, 0.72216603, 0.1607452 , 0.15471589, 0.18022921])
        info       = 0
        kw         = {'atol': 0, 'rtol': 1e-06}
        n          = 10
        rng        = Generator(PCG64) at 0x261DF1C8BA0
        solver     = <function tfqmr at 0x00000261B233D260>
        x          = array([ -1.22141228, -23.52876153,  -4.72398842, -21.36849916,
       -12.68970175,  -0.96738895,  32.85873013, -11.40287582,
        10.99363035,  30.93079899])
        x0         = array([0.63685093, 0.46836296, 0.0677924 , 0.52832101, 0.49467215,
       0.68333247, 0.3392967 , 0.57867322, 0.25729156, 0.60952728])
```

</details>

--- a/scipy/sparse/linalg/_eigen/tests/test_svds.py
+++ b/scipy/sparse/linalg/_eigen/tests/test_svds.py
@@ -405,16 +405,12 @@ class SVDSCommonTests:
         with pytest.raises(AssertionError, match=message):
             assert_equal(res1a, res1b)
 
-    @pytest.mark.parametrize("rng", (0, 1,
-                                     np.random.default_rng(0)))
-    def test_svd_rng_2(self, rng):
+    def test_svd_rng_2(self):
         n = 100
         k = 1
 
-        tmp_rng = np.random.default_rng(0)
-        A = tmp_rng.random((n, n))
-
-        rng = copy.deepcopy(rng)
+        rng = np.random.default_rng(234981)
+        A = rng.random((n, n))
         rng_2 = copy.deepcopy(rng)
 
         # with the same rng, solutions are the same and accurate
@@ -424,23 +420,20 @@ class SVDSCommonTests:
             assert_allclose(res1a[idx], res2a[idx], rtol=1e-15, atol=2e-16)
         _check_svds(A, k, *res1a)
 
-    @pytest.mark.parametrize("rng", (None,
-                                     np.random.default_rng(0)))
     @pytest.mark.filterwarnings("ignore:Exited",
                                 reason="Ignore LOBPCG early exit.")
-    def test_svd_rng_3(self, rng):
+    def test_svd_rng_3(self):
         n = 100
         k = 5
 
-        tmp_rng = np.random.default_rng(0)
-        A = tmp_rng.random((n, n))
-
-        rng = copy.deepcopy(rng)
+        rng1 = np.random.default_rng(0)
+        rng2 = np.random.default_rng(234832)
+        A = rng1.random((n, n))
 
         # rng in different state produces accurate - but not
         # not necessarily identical - results
-        res1a = svds(A, k, solver=self.solver, rng=rng, maxiter=1000)
-        res2a = svds(A, k, solver=self.solver, rng=rng, maxiter=1000)
+        res1a = svds(A, k, solver=self.solver, rng=rng1, maxiter=1000)
+        res2a = svds(A, k, solver=self.solver, rng=rng2, maxiter=1000)
         _check_svds(A, k, *res1a, atol=2e-7)
         _check_svds(A, k, *res2a, atol=2e-7)
 

--- a/scipy/sparse/linalg/_isolve/tests/test_iterative.py
+++ b/scipy/sparse/linalg/_isolve/tests/test_iterative.py
@@ -505,7 +505,7 @@ def test_x0_working(solver):
 
     x, info = solver(A, b, x0=x0, **kw)
     assert info == 0
-    assert norm(A @ x - b) <= 3e-6*norm(b)
+    assert norm(A @ x - b) <= 4.5e-6*norm(b)
 
 
 def test_x0_equals_Mb(case):

